core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,181 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
             learner_playlist_model.collection_ids if
             learner_playlist_model else [])
 
-    def test_mark_exploration_to_be_played_later(self):
+    def test_subscribed_exploration_cannot_be_added_to_playlist(self):
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, self.EXP_ID_0)
+
+        # Now if we try to add the same exploration, it shouldn't be added.
+        learner_progress_services.add_exp_to_learner_playlist(
+            self.user_id, self.EXP_ID_0)
         self.assertEqual(
             self._get_all_learner_playlist_exp_ids(self.user_id), [])
 
-        # Add an exploration to the learner playlist.
+    def test_single_exploration_is_added_correctly_to_playlist(self):
+        # Test adding a single exploration_id to learner playlist.
         learner_progress_services.add_exp_to_learner_playlist(
             self.user_id, self.EXP_ID_0)
         self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(self.user_id),
-            [self.EXP_ID_0])
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [self.EXP_ID_0])
+
+    def test_multiple_explorations_are_added_correctly_to_playlist(self):
+        # Test adding two explorations to the learner playlist.
+        learner_progress_services.add_exp_to_learner_playlist(
+            self.user_id, self.EXP_ID_0)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [self.EXP_ID_0])
 
-        # Add another exploration.
         learner_progress_services.add_exp_to_learner_playlist(
             self.user_id, self.EXP_ID_1)
         self.assertEqual(
             self._get_all_learner_playlist_exp_ids(
                 self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
-        # Add the first exploration in a different position.
+    def test_changing_order_of_added_explorations_has_no_playlist_effect(self):
```

Comment:
It seems to have an effect...the order is switched?

Reply:
It was going over the max line length but, got it!